presto-main/src/main/java/com/facebook/presto/operator/ExplainAnalyzeOperator.java

```diff
@@ -44,14 +45,16 @@
         private final PlanNodeId planNodeId;
         private final QueryPerformanceFetcher queryPerformanceFetcher;
         private final Metadata metadata;
+        private final CostCalculator costCalculator;
         private boolean closed;
 
-        public ExplainAnalyzeOperatorFactory(int operatorId, PlanNodeId planNodeId, QueryPerformanceFetcher queryPerformanceFetcher, Metadata metadata)
+        public ExplainAnalyzeOperatorFactory(int operatorId, PlanNodeId planNodeId, QueryPerformanceFetcher queryPerformanceFetcher, Metadata metadata, CostCalculator costCalculator)
```

Comment:
There would be cyclic dependency as `CostCalculator` implementation uses metadata. Maybe it should be refactored so the `CostCalculator` is created per transaction by `Metadata` implementation but I am not sure if that is correct approach. I will leave it as it is for now and let's discuss it offline.